src/views/bibliography_element.ts

```diff
@@ -89,6 +83,57 @@ export class BibliographyElementBlockView<
 
   public ignoreMutation = () => true
 
+  private handleEdit = (citationId: string) => {
+    this.props.popper.destroy() // destroy the context menu
+    this.showPopper(citationId)
+  }
+
+  private handleComment = (citationId: string) => {
+    this.props.setComment(buildComment(citationId) as CommentAnnotation)
+  }
+
+  public showContextMenu = (element: Element) => {
+    const componentProps: ContextMenuProps = {
+      actions: [
+        {
+          label: 'Edit',
+          action: () => this.handleEdit(element.id),
+          icon: 'EditIcon',
+        },
+        {
+          label: 'Comment',
+          action: () => this.handleComment(element.id),
+          icon: 'AddComment',
+        },
+      ],
+    }
+
+    this.editor = ReactSubView(
+      this.props,
+      ContextMenuWrapper,
+      componentProps,
+      this.node,
+      this.getPos,
+      this.view,
+      'context-menu'
+    )
+    this.props.popper.show(element, this.editor, 'right-start')
+  }
+
+  private onClickHandler = (element: Element, dataTracked?: TrackedAttrs) => {
+    this.elementWithContextMenu = null
+    this.props.popper.destroy() // destroy the old context menu
+    this.elementWithContextMenu = element
+    this.view.dispatch(this.view.state.tr.setMeta(CLEAR_SUGGESTION_ID, true))
```

Comment:
as dispatch may cause a rerender of a component and reevaluation of some of its methods it would be safer to combine all what you need to dispatch into a single transaction. So a more reliable approach would be to take a transaction (const newTr = this.view.state.tr) and apply all what you need on it and then dispatch it at the end of the function

Reply:
There is an update. dispach is called only if necessary...